Teapot/MockTeapot.swift

```diff
@@ -95,4 +117,20 @@ open class MockTeapot: Teapot {
             completion(nil, TeapotError.missingMockFile(resource))
         }
     }
+
+    private func checkHeadersAgainstExpected(headers: [String: String]?) -> Bool {
+        guard !headersToCheckFor.isEmpty else { /* nothing to check */ return true }
+
+        guard let received = headers else { /* We want to check but there's nothing to check */ return false }
```

Comment:
This comment should be something like `Headers expected but not received` to be more clear

Reply:
Oh and i think `receivedHeaders` as a var would make this easier to read.

Reply:
not a fan of inlined comments inside code like this. makes it harder to figure out where they start/end without syntax highlight.

Reply:
Fair enough, was trying to keep to the preference for one-line return statements. I will break these out.